extensions-contrib/gce-extensions/src/test/java/org/apache/druid/indexing/overlord/autoscaling/gce/GCEAutoScalerTest.java

```diff
@@ -0,0 +1,472 @@
+    EasyMock.verify(mockInstances);
+    EasyMock.verify(mockIdToIpRequest);
+  }
+
+  private InstanceGroupManagersListManagedInstancesResponse createRunningInstances(
+      List<String> instances
+  )
+  {
+    InstanceGroupManagersListManagedInstancesResponse mockResponse =
+        new InstanceGroupManagersListManagedInstancesResponse();
+    mockResponse.setManagedInstances(new ArrayList<>());
+    for (String x : instances) {
+      ManagedInstance mi = new ManagedInstance();
+      mi.setInstance(x);
+      mockResponse.getManagedInstances().add(mi);
+    }
+    return mockResponse;
+  }
+
+  @Test
+  public void testTerminateWithIds() throws IOException // for the mock calls, not really throwing
+  {
+    GCEAutoScaler autoScaler = new GCEAutoScaler(
+        2,
+        4,
+        new GCEEnvironmentConfig(1, "proj-x", "us-central-1", "druid-mig"),
+        null,
+        mockCompute // <-- I pretend to be Google
+    );
+
+    // set up getRunningInstances results
+    InstanceGroupManagersListManagedInstancesResponse beforeRunningInstance =
+        createRunningInstances(Arrays.asList(
+            "http://xyz/foo",
+            "http://xyz/bar",
+            "http://xyz/baz"
+        ));
+    InstanceGroupManagersListManagedInstancesResponse afterRunningInstance =
+        createRunningInstances(Arrays.asList(
+            "http://xyz/foo",
+            "http://xyz/bar"
+        ));
+
+    EasyMock.expect(mockInstancesRequest.execute()).andReturn(beforeRunningInstance); // 1st call
+    EasyMock.expect(mockInstancesRequest.setMaxResults(500L)).andReturn(mockInstancesRequest);
+    EasyMock.expect(mockInstancesRequest.execute()).andReturn(afterRunningInstance);  // 2nd call
+    EasyMock.expect(mockInstancesRequest.setMaxResults(500L)).andReturn(mockInstancesRequest);
+    EasyMock.replay(mockInstancesRequest);
+
+
+    EasyMock.expect(mockInstanceGroupManagers.listManagedInstances(
+            "proj-x",
+            "us-central-1",
+            "druid-mig"
+            )).andReturn(mockInstancesRequest).times(2);
+
+    // set up the delete operation
+    Operation mockResponse = new Operation();
+    mockResponse.setStatus("DONE");
+    mockResponse.setError(new Operation.Error());
+
+    EasyMock.expect(mockDeleteRequest.execute()).andReturn(mockResponse);
+    EasyMock.replay(mockDeleteRequest);
+
+    InstanceGroupManagersDeleteInstancesRequest requestBody =
+            new InstanceGroupManagersDeleteInstancesRequest();
+    requestBody.setInstances(Collections.singletonList("zones/us-central-1/instances/baz"));
+
+    EasyMock.expect(mockInstanceGroupManagers.deleteInstances(
+            "proj-x",
+            "us-central-1",
+            "druid-mig",
+            requestBody
+    )).andReturn(mockDeleteRequest);
+
+    EasyMock.replay(mockInstanceGroupManagers);
+
+    // called twice in getRunningInstances...
+    EasyMock.expect(mockCompute.instanceGroupManagers()).andReturn(mockInstanceGroupManagers);
```

Comment:
nit: can use `.times(3)` instead of repeating the expects

Reply:
This I prefer to keep unrolled for clarity as there are the comments explaining what is what